rust_src/src/windows.rs

```diff
@@ -636,6 +642,23 @@ impl LispObject {
     }
 }
 
+pub struct LispWindowIter {
+    current: Option<LispWindowRef>,
+}
+
+impl Iterator for LispWindowIter {
+    type Item = LispWindowRef;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.current {
+            None => None,
+            Some(window) => {
+                self.current = window.next.as_valid_window();
```

Comment:
The window is not checked for validity in the original `make_frame_visible_1`, so you can just call `.as_window()`